### ISSUE TITLE
java error when override geneFeatureToAlign

### ISSUE THREAD

**Issue by ZJURenyi** [milaboratory/mixcr#742]
## Checklist before submitting the issue:

 - [✓] The issue is strongly related to the MiXCR software
 - [✓] The issue can be reproduced with [the most recent version](https://github.com/milaboratory/mixcr/releases) of MiXCR
 - [✓] There is no answer to the question in the [official documentation](http://mixcr.readthedocs.io/) and there is no duplicate issue in the [bug tracker](https://github.com/milaboratory/mixcr/issues?utf8=✓&q=)  
 - [✓] Inspection of raw alignments with [exportAlignmentsPretty](http://mixcr.readthedocs.io/en/master/export.html#exporting-well-formatted-alignments-for-manual-inspection) shows that data has the expected architecture, and sample preparation artefacts are not the reason of the problem (if this is the matter of the issue)

## Expected Result

To analyse human TCRB from targeted multiplex PCR sequencing data

## Actual Result

When I set **-OvParameters.geneFeatureToAlign={L2Begin:VEnd}**, code ran ok. But if I changed to **FR3Begin:VEnd**, error occured as below

## Exact MiXCR commands 
code runs with no issue
```
mixcr align --species hs \
        -OvParameters.geneFeatureToAlign={L2Begin:VEnd} \
        raw_fq/${sample}_1.fq.gz raw_fq/${sample}_2.fq.gz result/${sample}.vdjca
```
code runs with **java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0**
```
mixcr align --species hs \
        -OvParameters.geneFeatureToAlign={FR3Begin:VEnd} \
        raw_fq/${sample}_1.fq.gz raw_fq/${sample}_2.fq.gz result/${sample}.vdjca
```

## MiXCR report files 

```
Alignment: 0%
picocli.CommandLine$ExecutionException: Error while running command (com.milaboratory.mixcr.cli.CommandAlign@724b939e): java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0
        at picocli.CommandLine.executeUserObject(CommandLine.java:1778)
        at picocli.CommandLine.access$900(CommandLine.java:145)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2141)
        at com.milaboratory.mixcr.cli.Main$1.handle(Main.java:84)
        at com.milaboratory.mixcr.cli.Main$1.handle(Main.java:73)
        at picocli.CommandLine$AbstractParseResultHandler.handleParseResult(CommandLine.java:1968)
        at com.milaboratory.mixcr.cli.Main.handleParseResult(Main.java:95)
        at com.milaboratory.mixcr.cli.Main.main(Main.java:67)
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0
        at com.milaboratory.core.sequence.AbstractArraySequence.codeAt(AbstractArraySequence.java:47)
        at com.milaboratory.core.sequence.NucleotideSequence.codeAt(NucleotideSequence.java:31)
        at com.milaboratory.core.alignment.kaligner1.KMapper.addReference(KMapper.java:349)
        at com.milaboratory.core.alignment.kaligner1.KAligner.addReference(KAligner.java:166)
        at com.milaboratory.core.alignment.kaligner1.KAligner.addReference(KAligner.java:137)
        at com.milaboratory.core.alignment.kaligner1.KAligner.addReference(KAligner.java:343)
        at com.milaboratory.core.alignment.kaligner1.KAligner.addReference(KAligner.java:44)
        at com.milaboratory.mixcr.vdjaligners.VDJCAlignerAbstract.createKAligner(VDJCAlignerAbstract.java:68)
        at com.milaboratory.mixcr.vdjaligners.VDJCAlignerAbstract.init(VDJCAlignerAbstract.java:171)
        at com.milaboratory.mixcr.vdjaligners.VDJCAligner.ensureInitialized(VDJCAligner.java:110)
        at com.milaboratory.mixcr.vdjaligners.VDJCAlignerWithMerge.init(VDJCAlignerWithMerge.java:57)
        at com.milaboratory.mixcr.vdjaligners.VDJCAligner.ensureInitialized(VDJCAligner.java:110)
        at com.milaboratory.mixcr.vdjaligners.VDJCAligner.process(VDJCAligner.java:68)
        at com.milaboratory.mixcr.vdjaligners.VDJCAligner.process(VDJCAligner.java:29)
        at cc.redberry.pipe.CUtils$9.process(CUtils.java:398)
        at cc.redberry.pipe.CUtils$9.process(CUtils.java:393)
        at cc.redberry.pipe.blocks.ParallelProcessor$Worker.run(ParallelProcessor.java:305)
        at java.base/java.lang.Thread.run(Thread.java:829)
```
I've also tested some other regions:
regions that work well:
VTranscript, VRegion, FR2
FR1Begin:FR2End
FR1Begin:FR3End
FR2Begin:FR3End

regions that report errors:
FR1Begin:FR1End (FR1)
```
picocli.CommandLine$ExecutionException: Error while running command (com.milaboratory.mixcr.cli.CommandAlign@2ab5afc7): java.lang.IllegalArgumentException: com.fasterxml.jackson.databind.JsonMappingException: Incorrect input: FR1 (through reference chain: com.milaboratory.mixcr.vdjaligners.VDJCAlignerParameters["vParameters"]->com.milaboratory.mixcr.vdjaligners.KGeneAlignmentParameters["geneFeatureToAlign"])
```
FR3Begin:FR3End (FR3)
```
picocli.CommandLine$ExecutionException: Error while running command (com.milaboratory.mixcr.cli.CommandAlign@96a75da): java.lang.IllegalArgumentException: com.fasterxml.jackson.databind.JsonMappingException: Incorrect input: FR3 (through reference chain: com.milaboratory.mixcr.vdjaligners.VDJCAlignerParameters["vParameters"]->com.milaboratory.mixcr.vdjaligners.KGeneAlignmentParameters["geneFeatureToAlign"])
```
CDR3Begin:VEnd
```
picocli.CommandLine$ExecutionException: Error while running command (com.milaboratory.mixcr.cli.CommandAlign@37854b34): java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0
```
CDR3
``` 
picocli.CommandLine$ExecutionException: Error while running command (com.milaboratory.mixcr.cli.CommandAlign@1d0a61c8): java.lang.IllegalArgumentException: com.fasterxml.jackson.databind.exc.ValueInstantiationException: Cannot construct instance of `com.milaboratory.mixcr.vdjaligners.VDJCAlignerParameters`, problem: `java.lang.IllegalArgumentException`
 at [Source: UNKNOWN; byte offset: #UNKNOWN]
```
any region with VEndTrimmed
```
picocli.CommandLine$ExecutionException: Error while running command (com.milaboratory.mixcr.cli.CommandAlign@149c3204): java.lang.IllegalArgumentException: Supports only pure basic reference points, VEndTrimmed is not basic.
```

**Comment by ZJURenyi** [comment]
After using imgt library, the issue is fixed.

**Comment by ZJURenyi** [comment]
Sorry but region with VEndTrimmed still got error

**Comment by PoslavskySV** [comment]
Thank you for reporting. Which MiXCR version you use?

**Comment by ZJURenyi** [comment]
> 

Hi, I'm using the version 4.0.0 from conda

**Comment by PoslavskySV** [comment]
Resolved in [MiXCR v4.1](https://github.com/milaboratory/mixcr/releases/tag/v4.1.0)
